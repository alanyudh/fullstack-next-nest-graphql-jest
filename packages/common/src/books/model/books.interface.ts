export interface IBook {
  id: string;
  title: string;
  description: string;
  year: number;
}
