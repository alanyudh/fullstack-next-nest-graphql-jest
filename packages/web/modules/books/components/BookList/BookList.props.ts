import { Book } from '../../models/book.model'

export interface BookListProps {
  books: Book[]
}
