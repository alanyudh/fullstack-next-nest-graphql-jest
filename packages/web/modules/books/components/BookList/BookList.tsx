import { List, Empty } from 'antd'
import BookListPreview from '../BookListPreview/BookListPreview'
import { BookListProps } from './BookList.props'

const BookList: React.FC<BookListProps> = ({ books }: BookListProps) => {
  if (!books || books.length < 1) return <Empty data-testid="book-list-empty" />

  return (
    <List
      grid={{
        gutter: 16,
        xs: 1,
        sm: 2,
        md: 4,
        lg: 4,
        xl: 6,
        xxl: 3,
      }}
      dataSource={books}
      renderItem={(item) => (
        <List.Item data-testid="book-list-item">
          <BookListPreview book={item} />
        </List.Item>
      )}
    />
  )
}

export default BookList
