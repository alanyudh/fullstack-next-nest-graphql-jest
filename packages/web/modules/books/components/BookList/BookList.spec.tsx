import { Book } from '../../models/book.model'
import { render } from '../../../../test/testUtils'
import { BookListProps } from './BookList.props'
import BookList from './BookList'

describe('BookList', () => {
  let expectedProps: BookListProps

  beforeEach(() => {
    expectedProps = {
      books: [
        new Book('1', 'title1', 'description1', 2001),
        new Book('2', 'title2', 'description2', 2002),
        new Book('3', 'title3', 'description3', 2003),
        new Book('4', 'title4', 'description4', 2004),
      ],
    }
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<BookList {...expectedProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render array of books', () => {
    const { getAllByTestId } = render(<BookList {...expectedProps} />)
    const bookPreviews = getAllByTestId('book-list-item')

    expect(bookPreviews).toHaveLength(expectedProps.books.length)
  })

  it('should render empty information when there is no books', () => {
    expectedProps.books = []

    const { getByTestId } = render(<BookList {...expectedProps} />)
    const emptyBox = getByTestId('book-list-empty')

    expect(emptyBox).toBeVisible()
  })
})
