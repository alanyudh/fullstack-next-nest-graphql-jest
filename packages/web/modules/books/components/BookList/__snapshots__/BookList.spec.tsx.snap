// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-list ant-list-split ant-list-grid"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-row"
          style="margin-left: -8px; margin-right: -8px;"
        >
          <div>
            <div
              class="ant-col"
              style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
            >
              <div
                class="ant-list-item"
                data-testid="book-list-item"
              >
                <a
                  data-testid="book-list-anchor"
                  href="/books/1"
                >
                  <div
                    class="ant-card ant-card-bordered"
                  >
                    <div
                      class="ant-card-head"
                    >
                      <div
                        class="ant-card-head-wrapper"
                      >
                        <div
                          class="ant-card-head-title"
                        >
                          title1
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-card-body"
                    >
                      description1
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
            >
              <div
                class="ant-list-item"
                data-testid="book-list-item"
              >
                <a
                  data-testid="book-list-anchor"
                  href="/books/2"
                >
                  <div
                    class="ant-card ant-card-bordered"
                  >
                    <div
                      class="ant-card-head"
                    >
                      <div
                        class="ant-card-head-wrapper"
                      >
                        <div
                          class="ant-card-head-title"
                        >
                          title2
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-card-body"
                    >
                      description2
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
            >
              <div
                class="ant-list-item"
                data-testid="book-list-item"
              >
                <a
                  data-testid="book-list-anchor"
                  href="/books/3"
                >
                  <div
                    class="ant-card ant-card-bordered"
                  >
                    <div
                      class="ant-card-head"
                    >
                      <div
                        class="ant-card-head-wrapper"
                      >
                        <div
                          class="ant-card-head-title"
                        >
                          title3
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-card-body"
                    >
                      description3
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div>
            <div
              class="ant-col"
              style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
            >
              <div
                class="ant-list-item"
                data-testid="book-list-item"
              >
                <a
                  data-testid="book-list-anchor"
                  href="/books/4"
                >
                  <div
                    class="ant-card ant-card-bordered"
                  >
                    <div
                      class="ant-card-head"
                    >
                      <div
                        class="ant-card-head-wrapper"
                      >
                        <div
                          class="ant-card-head-title"
                        >
                          title4
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-card-body"
                    >
                      description4
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
